mantidproject/mantid
# Allow blocking of signals to the AlgorithmProgress

**Description of work.**
Allows blocking of all signals to the AlgorithmProgressPresenter. This stops the progress bar from updating. Use case is to prevent updates to the progress bar during ProjectRecovery, which uses a separate progress bar.

**To test:**
- Enable Workbench project recovery
- Make lots of workspaces, you could use this:
```python
for i in range(100):
    CreateSampleWorkspace(OutputWorkspace="test{}".format(i))
```
- Wait for project recovery to save (I changed it to 5s save & restarted workbench for it to be in effect)
- Crash the Workbench with `Segfault(0)`
- Reopen Workbench, accept project recovery
  - The Workbench progress bar shouldn't move
  - The Project Recovery progress bar should be the only one moving



Fixes #25324




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
